tests/nativeapp/test_new_sql_facade.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import deque
+from textwrap import dedent
+from typing import Any, List, Tuple
+
+from snowflake.cli._plugins.nativeapp.sf_sql_facade import SnowflakeSQLFacade
+from snowflake.cli.api.sql_execution import SqlExecutor
+
+# Objective:
+# - correct calls are made to SF. in the correct order.
+# Requirements:
+
+# - can ensure execute_queries is called with expected query (closest method to SF before _execute_string, all queries call this method.)
+# - can mock execute_queries to return result we want
+# - can assert calls are made in the right order
+
+
+class MockQueriesExecutor(SqlExecutor):
+    def __init__(self, expected_calls: List[Tuple[str, Any]]):
+        super().__init__()
+        self._mock_results = deque(expected_calls)
+
+    @property
+    def _conn(self):
+        raise Exception("Connection should not be accessed in tests.")
+
+    def _execute_queries(self, queries: str, **kwargs):
+        (query_expected, result) = self._mock_results.popleft()
+        if dedent(queries).strip() != dedent(query_expected).strip():
+            raise Exception(
+                f"Test failed, expected query {query_expected} but received query {queries}"
+            )
```

Comment:
```suggestion
        assert dedent(queries).strip() == dedent(query_expected).strip(), f"Expected query {query_expected} but received query {queries}"
```
that way we get pytest assertion rewriting prettiness